Landing page created. Resolves #187

LowLifeArcade opened the pull request:
I created a landing page with a hero section on top and a testimonial section below that. 🥳 

@audreyfeldroy here's a list of what remains:
- [x] Split up landing-page.js into separate files for the components. 
- [x] Convert media queries to use tailwind utility classes for breakpoints.
- [ ] Replace placeholder text with real copy.
- [ ] Change site homepage to be landing-page.js

Resolves #187

RedFox0x20 commented on pages/landing-page.js:
```diff
@@ -0,0 +1,437 @@
+        <div className="testimonial-card">
+          <div className="card glass lg:card text-neutral-content">
+            <figure className="p-0">
+              <Image
+                alt="testimonial"
+                className="shadow-lg"
+                height={340}
+                layout="responsive"
+                src="/images/HealthWorkShop.jpg"
+                width={500}
+              />
```
A quick pointer regarding images, it's ideal to import the image as it automatically provides the correct width and height, ensuring the image looks as good as possible.

```js
import HealthWorkshopImage from '/images/HealthWorkshop.jpg';
...
src={HealthWorkshopImage}
```

RedFox0x20 commented on pages/landing-page.js:
```diff
@@ -0,0 +1,437 @@
+              }
+            </p>
+
+            <button
+              className="btn btn-accent rounded-full shadow-lg"
+              onClick={() => router.push('/in-kind')}
+              type="button"
+            >
+              {'Help Today'}
+            </button>
```
Perhaps instead of a `button`, a `Link` from NextJS can be used instead.

audreyfeldroy replied:
That's a good idea. @LowLifeArcade check out the Next.js `Link` docs at https://nextjs.org/docs/api-reference/next/link and let us know if you need any help figuring it out 😄

RedFox0x20 reviewed (commented):
@LowLifeArcade Thanks for the contributions, I can see that you have left plenty of notes of things that are still a work in progress and listed these things out, as such I have marked this PR as a Draft while it is worked on.

I'm liking the look of this page, it's very tidy in particular the testimonial section!

Just to add to your list of things to tweak, I believe it would be ideal to have the page share the same basic look and feel as the rest of the site. In particular I'm looking at the navbar which has a completely different look to it.

Also to avoid the `<style>` tag where possible, use of the Tailwind and DaisyUI classes is very much preferred and for almost every change made in the `<style>` can be done using `className`, this will likely occur naturally as elements are broken up into smaller components
> Split up landing-page.js into separate files for the components.

Finally in regards to the following, perhaps moving `index.js` to something more suitable such as `home.js` and subsequently moving `landing-page.js` to `index.js` would be a suitable option for this.
> Change site homepage to be landing-page.js

audreyfeldroy commented on pages/landing-page.js:
```diff
@@ -0,0 +1,437 @@
+ * @param {String} [props.opacity = '60' ]"bg-opacity-`opacity`" | Example: `"30"` | [Tailwind opacity](https://tailwindcss.com/docs/background-opacity)
+ * @param {*} props.children - Children elements
+ * @returns {ReactElement} JSX
+ */
```
This is really cool @LowLifeArcade!

audreyfeldroy commented on pages/landing-page.js:
```diff
@@ -0,0 +1,437 @@
+import PropTypes from 'prop-types';
+import { useRouter } from 'next/router';
+
+// import { BsSearch } from 'react-icons/bs';
```
Hey, I'm curious why you decided to put this in, then comment it out? I vaguely remember it being in the navbar?

RedFox0x20 replied:
That element has been commented out, in addition to a few other elements in the navbar which here is a copy of the original code with redactions. Depending on the decisions made with the navbar here, I may modify the original to make it more suitable to these broader cases where elements may want to be added/remove per page.

audreyfeldroy commented on pages/landing-page.js:
```diff
@@ -0,0 +1,437 @@
+      <main className="body">
+        <LargeHeroSection
+          bgImage="/images/HotMealDay.jpg"
+          heroHeight="85vh"
```
```suggestion
```

Let's try removing this to let `heroHeight be the default value from line 36.

audreyfeldroy reviewed (changes requested):
This is great @LowLifeArcade, thanks for submitting it! Try out the "Commit Suggestion" button to see what it does, and we can continue to work on this PR together over screenshare. 😄

LowLifeArcade commented:
> @LowLifeArcade Thanks for the contributions, I can see that you have left plenty of notes of things that are still a work in progress and listed these things out, as such I have marked this PR as a Draft while it is worked on.

Yeah!

> Just to add to your list of things to tweak, I believe it would be ideal to have the page share the same basic look and feel as the rest of the site. In particular I'm looking at the navbar which has a completely different look to it.

I hear you on that. The navbar is an interesting point as I am trying to figure out how it fits in. As per the referred [blog post](https://unbounce.com/landing-page-examples/high-converting-landing-pages/) there's a push to almost omit the whole thing. @audreyfeldroy had stated she wanted to keep it in, so I modified it to meet the recommendations. Stylistically ,going with the blog post, it made sense that it was less observable and to really only appear as branding in my mind, so I went with an opaque bar with minimal content. Of course we're all still in conversation about these things, so let's keep talking and sharing thoughts.

> Also to avoid the `<style>` tag where possible, use of the Tailwind and DaisyUI classes is very much preferred and for almost every change made in the `<style>` can be done using `className`, this will likely occur naturally as elements are broken up into smaller components

I took this into account as well as with a conversation with @audreyfeldroy and refactored the code using tailwind's utility classes. 

> > Split up landing-page.js into separate files for the components.

Definitely!

> Finally in regards to the following, perhaps moving `index.js` to something more suitable such as `home.js` and subsequently moving `landing-page.js` to `index.js` would be a suitable option for this.
> > Change site homepage to be landing-page.js

I believe @audreyfeldroy has a plan or idea there.

audreyfeldroy commented:
Merging this so we can iterate further during Hacktoberfest, breaking remaining to-do's into small, easy ones for beginners to contribute to. The landing page is still only hidden at /landing-page so it shouldn't affect the rest of the site. Thanks for all your work here @LowLifeArcade and for your thorough and thoughtful reviewing @RedFox0x20 🙏
